sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -207,6 +207,26 @@ trait CheckAnalysis extends PredicateHelper {
                 s"of type ${condition.dataType.catalogString} is not a boolean.")
 
           case Aggregate(groupingExprs, aggregateExprs, child) =>
+            val distinctAggregateExprs = aggregateExprs.flatMap(_.collect {
+              case ae: AggregateExpression if ae.isDistinct => ae
+            })
+            val distinctAggGroups = distinctAggregateExprs.groupBy { e =>
+              val unfoldableChildren = e.aggregateFunction.children.filter(!_.foldable).toSet
+              if (unfoldableChildren.nonEmpty) {
+                unfoldableChildren
+              } else {
+                e.aggregateFunction.children.take(1).toSet
```

Comment:
Since this is duplicate code with `RewriteDistinctAggregates`, how about defining a helper function to extract distinct aggregate groups? e.g., `RewriteDistinctAggregates.extractDistinctAggregates`. I think it is difficult to understand this logic without [the comments](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala#L177-L183).

Reply:
This PR is changed to implement multiple DISTINCT aggregate with Filter clause. I modified the title and description of this PR.